web/permissions/permissions.go

```diff
@@ -126,7 +127,23 @@ func createPermission(c echo.Context) error {
 		}
 	}
 
-	pdoc, err := permission.CreateShareSet(instance, parent, sourceID, codes, shortcodes, subdoc.Permissions, expiresAt)
+	// Handles the metadata part
+	md := metadata.New()
+	md.CreatedByApp = sourceID
+	md.DocTypeVersion = permission.DoctypeVersion
+	err = md.UpdatedByApp(sourceID, "") // We do not have a version here
+	if err != nil {
+		return err
+	}
+
+	// Adding metadata if it does not exist
+	if subdoc.Metadata == nil {
+		subdoc.Metadata = md
+	} else { // Otherwise, ensure we have all the needed fields
+		subdoc.Metadata.EnsureCreatedFields(md)
```

Comment:
It looks like you are changing what this route accepts in parameter. You should update documentation if it is the case.